pydy/tests/test_system.py

```diff
@@ -395,3 +397,67 @@ def test_integrate(self):
         sys = System(self.kane, times=times)
         with testing.assert_raises(NotImplementedError):
             sys.generate_ode_function(generator='made-up')
+
+    def test_joint_system_init(self):
+        sys = System(method='joints')
```

Comment:
This made me think of an idea. So we already have `Methods` classes. A method essentially takes a collection of RigidBodies, Particles, and Forces/Torques and outputs the symbolic equations of motion. The RigidBodies and Particles already have their kinematics prescribed before they arrive to the Method class. You're new system construction stuff could be considered a Method too. Except your your Method requires, Bodies, Particles, Forces/Torques, and Joints where the kinematics are not prescribed for the RigidBodies and Particles, but instead you pass in Joints. But inside your Method, you first use the Joints to prescribe the kinematics and then use another Method to finally compute the symbolic EoMs.

So the point I'm getting at is that you could create a new Method class called `JointMethod` or something that housed the logic of prescribing the kinematics to the RigidBodies and Particles. This would mean that System doesn't really have to change. We would still initialize a System with a method class.

This is a derailment from the current ideas you've developed, but maybe one worth pondering. It just occurred to me.